packages/react-ui-core/src/Collapsible/Collapsible.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Button } from '../Button'
+
+const noop = () => {}
+
+export default class Collapsible extends Component {
+
+  constructor(props) {
+    super(props)
+    this.handleClick = this.handleClick.bind(this)
+    this.state = {
+      display: this.props.visible,
+    }
+  }
+
+  handleClick() {
+    this.props.handleClick()
+    this.setState({
+      display: !this.state.display,
+    })
+  }
+
+  render() {
+    const {
+      id,
+      theme,
+      showableItems,
+      nonshowableItems,
+      hiddenText,
+      visibleText,
+    } = this.props
+
+    return (
+      <div id={id} className={theme.collapsible}>
+        {showableItems}
+        <div className={this.state.display ? theme.show : theme.hide}>
+          {nonshowableItems}
+        </div>
+        <div className={theme.alignBottom}>
+          <Button className={theme.strong} onClick={this.handleClick}>
+            {this.state.display ? visibleText : hiddenText}
+          </Button>
+        </div>
+      </div>
+    )
+  }
+}
+
+Collapsible.defaultProps = {
+  showableItems: '',
+  nonshowableItems: '',
+  id: '',
+  title: '',
+  hiddenText: 'show more',
+  visibleText: 'show less',
+  visible: false,
+  handleClick: noop,
+}
```

Comment:
you probably want to add a `theme` to `propTypes` and set the default prop of `theme` to `{}`. I.e.:

```javascript
static propTypes = {
  ...
  theme: PropTypes.object,
}

static defaultProps = {
  ...
  theme: {},
}
```